src/Accounts/Authenticators/AccessTokenAuthenticator.cs

```diff
@@ -41,21 +41,27 @@ public override Task<IAccessToken> Authenticate(AuthenticationParameters paramet
             };
 
             if ((resourceId.EqualsInsensitively(environment.AzureKeyVaultServiceEndpointResourceId) ||
-                 resourceId.EqualsInsensitively(AzureEnvironment.Endpoint.AzureKeyVaultServiceEndpointResourceId))
+                 resourceId.EqualsInsensitively(AzureEnvironment.Endpoint.AzureKeyVaultServiceEndpointResourceId) ||
```

Comment:
So, it looks like the common change in behavior is to allow the resourceID parameter to either match the actual resourceId, or to match the endpoint name that contains the resource ID, which is the correct behavior.

Alternately, we could normalize the AuthenticationParameters to always use one form or the other.

Either way, we need a regression test in place to ensure we always do this.  How difficult would it be to write a unit test for this, or add this to an existing unit test?